src/test/regress/expected/pg17_json.out

```diff
@@ -0,0 +1,578 @@
+-- JSON_TABLE can be in the outer part of the join
+-- as long as there is a distributed table
+SELECT sub.*
+FROM my_films JOIN
+     lateral
+  (SELECT *
+   FROM JSON_TABLE (js, '$.favorites[*]' COLUMNS (id FOR ORDINALITY,
+                            kind text PATH '$.kind', NESTED PATH '$.films[*]'
+                            COLUMNS (title text PATH '$.title', director text PATH '$.director'))) AS jt
+   LIMIT 1000) AS sub ON (true) LEFT JOIN test_table ON(my_films.id = test_table.id)
+    ORDER BY 1,2,3,4;
+ id |   kind   |      title       |     director
+---------------------------------------------------------------------
+  1 | comedy   | Bananas          | Woody Allen
+  1 | comedy   | Bananas2         | Woody Allen
+  1 | comedy   | The Dinner Game  | Francis Veber
+  1 | comedy   | The Dinner Game2 | Francis Veber
+  2 | horror   | Psycho           | Alfred Hitchcock
+  2 | horror   | Psycho2          | Alfred Hitchcock
+  3 | thriller | Vertigo          | Alfred Hitchcock
+  3 | thriller | Vertigo2         | Alfred Hitchcock
+  4 | drama    | Yojimbo          | Akira Kurosawa
+  4 | drama    | Yojimbo2         | Akira Kurosawa
+(10 rows)
+
+-- JSON_TABLE can be on the outer side of the join
+-- We support outer joins where the outer rel is a recurring one
+-- and the inner one is a non-recurring one if we don't reference the outer from the inner
+-- https://github.com/citusdata/citus/pull/6512
+SELECT *
+FROM json_table('[{"a":10,"b":20},{"a":30,"b":40}]'::JSONB, '$[*]'
+               COLUMNS (id FOR  ORDINALITY, column_a int4 PATH '$.a', column_b int4 PATH '$.b', a int4, b int4, c text))
+LEFT JOIN LATERAL
+  (SELECT *
+   FROM my_films) AS foo on(foo.id = a);
+DEBUG:  recursively planning right side of the left join since the outer side is a recurring rel
+DEBUG:  recursively planning the distributed subquery since it is part of a distributed join node that is outer joined with a recurring rel
+DEBUG:  generating subplan XXX_1 for subquery SELECT id, js FROM pg17_json.my_films
+DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT "json_table".id, "json_table".column_a, "json_table".column_b, "json_table".a, "json_table".b, "json_table".c, foo.id, foo.js FROM (JSON_TABLE('[{"a": 10, "b": 20}, {"a": 30, "b": 40}]'::jsonb, '$[*]' AS json_table_path_0 COLUMNS (id FOR ORDINALITY, column_a integer PATH '$."a"', column_b integer PATH '$."b"', a integer PATH '$."a"', b integer PATH '$."b"', c text PATH '$."c"')) LEFT JOIN LATERAL (SELECT intermediate_result.id, intermediate_result.js FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(id bigint, js jsonb)) foo ON ((foo.id OPERATOR(pg_catalog.=) "json_table".a)))
+ id | column_a | column_b | a  | b  | c | id | js
+---------------------------------------------------------------------
+  1 |       10 |       20 | 10 | 20 |   |    |
+  2 |       30 |       40 | 30 | 40 |   |    |
+(2 rows)
```

Comment:
thank for adjusting tests for https://github.com/citusdata/citus/pull/6512

Reply:
Thank you for adding support for outer joins where the outer rel is a recurring one and the inner one is a non-recurring one if we don't reference the outer from the inner :D